### ISSUE TITLE
Declaring directives in select2 formatSelectionTooBig

### ISSUE THREAD

**Issue by frapontillo** [angular-ui/angular-ui-OLDREPO#286]
Hello, I am using the `ui-select2` directive in my app, but there's a problem with directives declared in the string returned by the `formatSelectionTooBig` function, as they don't fire (of course, they work in regularly written DOM elements).

For example, I have the following function:

``` javascript
$scope.formatSelectionTooBig = function(max) {
        return "<span translate>You can only select</span> " +
            max +
            ((max > 1 ? " <span translate>groups</span>" : " <span translate>group</span>")) +
            ".";
    };
```

where the `translate` directive can be found [here](https://github.com/StephanHoyer/ng-translate) (any other of my directives won't fire as well).

Is there a workaround/fix for this? If I set up a filter, returning something like `<span translate>{{'You can only select' | myFilter}}</span>` it doesn't work either.

Thank you.

**Comment by ProLoser** [comment]
This is injected via Select2, ergo it's outside of the AngularJS scope.

You _could_ try compiling it manually before returning but I am not sure exactly how it will be injected or if it's safe from future bugs.

Instead, I would recommend separating out the translation-specific logic into a filter or standalone function (if possible) and running it on the string BEFORE returning it to Select2. This would be the most efficient solution and would probably prove to be the most error-free in the long run.

...

Reviewing your ngTranslate code, looks like you could simply inject the translate service into your controller and do this.

**Off Topic:** I would have suggested opening a pull request with your translate directive, but it does not look that efficient. You may want to open a thread on the Google Groups asking for review/feedback.

**Comment by frapontillo** [comment]
Thank you for your reply. `ngTranslate` was not written by me, I'm using it with some editing, but it still is separated into service and directive. In fact, that's what I'm doing: I have injected the translate service into the controller and used it to translate the string before returning it to select2.

Still, I would have preferred to keep all of my translation logic in the DOM, so I can quickly see what needs to be translated without having to review my code as well.

What sounds strange is that if Select2 injects the string component into the DOM, then why doesn't Angular realizes that a new element has been added and compiles it, so to run the custom directives/filters?
